stack-graphs/src/c.rs

```diff
@@ -2035,3 +2054,38 @@ pub extern "C" fn sg_forward_partial_path_stitcher_free(
 ) {
     drop(unsafe { Box::from_raw(stitcher as *mut InternalForwardPartialPathStitcher) });
 }
+
+//-------------------------------------------------------------------------------------------------
+// Cancellation
+
+struct AtomicUsizeCancellationFlag<'a>(Option<&'a AtomicUsize>);
+impl CancellationFlag for AtomicUsizeCancellationFlag<'_> {
+    fn check(&self, at: &'static str) -> Result<(), crate::CancellationError> {
+        self.0
+            .map(|flag| {
+                if flag.fetch_and(0b0, std::sync::atomic::Ordering::Relaxed) != 0 {
+                    Err(CancellationError(at))
+                } else {
+                    Ok(())
+                }
+            })
+            .unwrap_or(Ok(()))
+    }
+}
+
+/// Describes the result of a computation
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum sg_result {
+    Success,
+    Cancelled,
```

Comment:
These should be `SG_RESULT_SUCCESS` and `SG_RESULT_CANCELLED` so that they conform to C's naming/capitalization conventions instead of Rust's